### PR TITLE
fix: address PR 41 review comments

### PR DESCRIPTION
**Pull request by sansenjian**
## Summary

- serialize rank detail payloads, deduplicate song IDs, and cap MID resolution at five concurrent requests
- isolate explicit request cookies, prevent QR login from mutating the process-global session, and validate favorite-MV pagination
- validate JSON/JSONP responses, harden smoke retries and stdout parsing, and correct API metadata and documentation
- move Explorer startup into a guarded bootstrap entry and preserve native form validation for Ctrl/Cmd+Enter

## Review notes

The CodeRabbit duplicate-declaration finding in tests/integration/api/readonly-extras.test.ts was verified against origin/dev and is a false positive; the reported duplicate declarations are not present.

## Validation

- npm run lint
- npm run build
- npm run test (55 files, 545 tests)
- npm run docs:build
- npm run smoke:e2e
- git diff --check

## Summary by Sourcery

改进多个用户及元数据 API 中的排行解析、用户会话处理、API Explorer 启动方式以及响应校验。

新特性：
- 添加受保护的 Explorer 引导入口，仅在所需的 DOM 和元数据存在时才启动浏览器端应用。

缺陷修复：
- 在 `getRanks` 中将歌曲详情请求序列化为 JSON 字符串，去重歌曲 ID，并将 `resolveMid` 的并发详情查询数量限制为 5 个。
- 防止二维码登录检查修改全局默认用户会话，同时仍支持显式选择持久化的客户端。
- 确保显式的 `userMusicu` 请求只使用其提供的 Cookie 作为凭据，避免从全局会话中“泄漏”身份信息。
- 校验 `getUserFavMv` 的分页入参，拒绝非正数或非整数的 page/limit 值。
- 强化 `y_common` 的 JSON/JSONP 处理，将格式错误的响应视为失败，并使用不同的 referer 进行重试。
- 使 JSON-RPC stdout 客户端在遇到非 JSON 行时更加健壮，忽略解析错误而不是崩溃。
- 在通过 Ctrl/Cmd+Enter 提交 API Explorer 请求时保留原生表单校验。
- 更新用户头像尺寸选项的 API 元数据，加入示例值并确保测试覆盖头像尺寸相关元数据。

改进：
- 明确排行 API 文档中关于 `resolveMid` 行为、去重策略和并发限制的说明。
- 优化冒烟测试 HTTP 服务器的等待循环，即使在成功获取但返回非 OK 响应时也始终在轮询之间延迟。
- 调整 Explorer 响应的样式，对长内容使用更健壮的自动换行方式。

构建：
- 将 Explorer 的构建入口和生成的 banner 从 `explorerApp` 切换到 `explorerBootstrap`，以获得更安全的启动语义。

测试：
- 扩展 `getRanks`、`userMusicu`、`y_common`、`getUserFavMv`、API 元数据、`checkQQLoginQr` 以及 API Explorer 的单元与集成测试，以覆盖新行为和边缘情况。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Improve rank resolution, user session handling, API explorer startup, and response validation across multiple user and metadata APIs.

New Features:
- Add a guarded Explorer bootstrap entry that only starts the browser app when required DOM and metadata are present.

Bug Fixes:
- Serialize song detail requests as JSON strings, deduplicate song IDs, and cap resolveMid processing to five concurrent detail lookups in getRanks.
- Prevent QR login checks from mutating the global default user session while still supporting clients that opt into persistence.
- Ensure explicit userMusicu requests use only their provided cookie for credentials, avoiding leakage from the global session.
- Validate getUserFavMv pagination inputs and reject non-positive or non-integer page/limit values.
- Harden y_common JSON/JSONP handling by treating malformed responses as failures and retrying with a different referer.
- Make JSON-RPC stdout client robust to non-JSON lines by ignoring parse errors instead of crashing.
- Preserve native form validation when submitting API Explorer requests via Ctrl/Cmd+Enter.
- Update API metadata for user avatar size options to include example values and ensure tests cover avatar size metadata.

Enhancements:
- Clarify rank API documentation around resolveMid behavior, deduplication, and capped concurrency.
- Refine smoke test HTTP server wait loop to always delay between polls, even on successful fetch attempts that return non-OK responses.
- Adjust Explorer response styling to use more resilient word wrapping for long content.

Build:
- Switch Explorer build entry point and generated banner from explorerApp to explorerBootstrap for safer startup semantics.

Tests:
- Expand unit and integration tests for getRanks, userMusicu, y_common, getUserFavMv, API metadata, checkQQLoginQr, and the API Explorer to cover new behaviors and edge cases.

</details>

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

改进多个用户及元数据 API 中的排行解析、用户会话处理、API Explorer 启动方式以及响应校验。

新特性：
- 添加受保护的 Explorer 引导入口，仅在所需的 DOM 和元数据存在时才启动浏览器端应用。

缺陷修复：
- 在 `getRanks` 中将歌曲详情请求序列化为 JSON 字符串，去重歌曲 ID，并将 `resolveMid` 的并发详情查询数量限制为 5 个。
- 防止二维码登录检查修改全局默认用户会话，同时仍支持显式选择持久化的客户端。
- 确保显式的 `userMusicu` 请求只使用其提供的 Cookie 作为凭据，避免从全局会话中“泄漏”身份信息。
- 校验 `getUserFavMv` 的分页入参，拒绝非正数或非整数的 page/limit 值。
- 强化 `y_common` 的 JSON/JSONP 处理，将格式错误的响应视为失败，并使用不同的 referer 进行重试。
- 使 JSON-RPC stdout 客户端在遇到非 JSON 行时更加健壮，忽略解析错误而不是崩溃。
- 在通过 Ctrl/Cmd+Enter 提交 API Explorer 请求时保留原生表单校验。
- 更新用户头像尺寸选项的 API 元数据，加入示例值并确保测试覆盖头像尺寸相关元数据。

改进：
- 明确排行 API 文档中关于 `resolveMid` 行为、去重策略和并发限制的说明。
- 优化冒烟测试 HTTP 服务器的等待循环，即使在成功获取但返回非 OK 响应时也始终在轮询之间延迟。
- 调整 Explorer 响应的样式，对长内容使用更健壮的自动换行方式。

构建：
- 将 Explorer 的构建入口和生成的 banner 从 `explorerApp` 切换到 `explorerBootstrap`，以获得更安全的启动语义。

测试：
- 扩展 `getRanks`、`userMusicu`、`y_common`、`getUserFavMv`、API 元数据、`checkQQLoginQr` 以及 API Explorer 的单元与集成测试，以覆盖新行为和边缘情况。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Improve rank resolution, user session handling, API explorer startup, and response validation across multiple user and metadata APIs.

New Features:
- Add a guarded Explorer bootstrap entry that only starts the browser app when required DOM and metadata are present.

Bug Fixes:
- Serialize song detail requests as JSON strings, deduplicate song IDs, and cap resolveMid processing to five concurrent detail lookups in getRanks.
- Prevent QR login checks from mutating the global default user session while still supporting clients that opt into persistence.
- Ensure explicit userMusicu requests use only their provided cookie for credentials, avoiding leakage from the global session.
- Validate getUserFavMv pagination inputs and reject non-positive or non-integer page/limit values.
- Harden y_common JSON/JSONP handling by treating malformed responses as failures and retrying with a different referer.
- Make JSON-RPC stdout client robust to non-JSON lines by ignoring parse errors instead of crashing.
- Preserve native form validation when submitting API Explorer requests via Ctrl/Cmd+Enter.
- Update API metadata for user avatar size options to include example values and ensure tests cover avatar size metadata.

Enhancements:
- Clarify rank API documentation around resolveMid behavior, deduplication, and capped concurrency.
- Refine smoke test HTTP server wait loop to always delay between polls, even on successful fetch attempts that return non-OK responses.
- Adjust Explorer response styling to use more resilient word wrapping for long content.

Build:
- Switch Explorer build entry point and generated banner from explorerApp to explorerBootstrap for safer startup semantics.

Tests:
- Expand unit and integration tests for getRanks, userMusicu, y_common, getUserFavMv, API metadata, checkQQLoginQr, and the API Explorer to cover new behaviors and edge cases.

</details>

</details>

### CODE REVIEW

File: tests/unit/services/apis/user/userMusicu.test.ts [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
+	test('does not mix default-session credentials into an explicit cookie', async () => {
+		await callUserMusicu({
+			module: 'test.module',
+			method: 'TestMethod',
+			param: {},
+			cookie: 'uin=request-user',
+		});
+
+		const payload = JSON.parse(requestMock.mock.calls[0][0].options.data);
+		expect(payload.comm).toMatchObject({ uin: 'request-user', loginUin: 'request-user' });
```
[the remark this answers is not in the document]
Comment (sansenjian) [reply tests/unit/services/apis/user/userMusicu.test.ts:35]: Implemented in b66f174. Added coverage for the implicit default-session path and asserted uin, loginUin, authst, and g_tk derived from the stored cookie.

File: tests/unit/explorer/explorerApp.test.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { getDeepLinkParams, initExplorerApp } from '../../../src/explorer/explorerApp';
+
+describe('explorer/explorerApp', () => {
+	test('can be imported without starting the browser app', () => {
+		expect(initExplorerApp).toEqual(expect.any(Function));
+		expect(getDeepLinkParams).toEqual(expect.any(Function));
+	});
```
[the remark this answers is not in the document]
Comment (sansenjian) [reply tests/unit/explorer/explorerApp.test.ts:7]: Implemented in b66f174. Added guarded bootstrap tests for missing document, missing request-form, missing metadata script, and the valid startup path.

File: docs/api/rank.md [hunk 3]
```diff
@@ -88,15 +90,15 @@ curl "http://localhost:3200/getSongInfo?songid=123456"
...
 
 ```bash
 curl "http://localhost:3200/getRanks?topId=4&limit=20&resolveMid=true"
 ```
 
-服务端会对列表里每首缺少 `mid` 的歌曲额外调用一次详情接口，把 `song_mid` / `mid` 字段补上。
+服务端会按 `songId` 去重，并以最多 5 个并发请求查询缺少 `mid` 的歌曲详情，再补全 `song_mid` / `mid` 字段。
 
 **优点：** 调用方便，一步到位  
 **缺点：** N 首歌多 N 次请求，榜单较大时会明显变慢。请谨慎使用。
```
[the remark this answers is not in the document]
Comment (sansenjian) [reply docs/api/rank.md]: Updated in b66f174. The drawback now refers to the deduplicated number of songs missing mid instead of claiming every song always adds a request.

File: tests/unit/services/apis/user/userMusicu.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sansenjian) [reply tests/unit/services/apis/user/userMusicu.test.ts:35]: Implemented in b66f174. Added the complementary no-explicit-cookie test for the default session, including authentication fields derived from the stored cookie.

File: docs/api/rank.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sansenjian) [reply docs/api/rank.md]: Updated in b66f174. The documentation now matches the deduplicated, bounded N+1 behavior.

File: scripts/smoke-mcp-explorer.ts [hunk 4]
```diff
@@ -175,7 +175,12 @@ class JsonRpcStdioClient {
 				continue;
 			}
 
-			const response = JSON.parse(line) as JsonRpcResponse;
+			let response: JsonRpcResponse;
+			try {
+				response = JSON.parse(line) as JsonRpcResponse;
+			} catch {
+				continue;
+			}
```
[the remark this answers is not in the document]
Comment (sansenjian) [reply scripts/smoke-mcp-explorer.ts]: Fixed in b66f174. Stdout parsing now accepts only JSON-RPC 2.0 objects with a safe integer id; null, arrays, scalars, empty objects, and malformed lines are ignored. Added focused regression tests.

File: src/services/apis/y_common.ts [hunk 5]
```diff
@@ -70,13 +71,8 @@ function looksValid(response: unknown): boolean {
...
 		const trimmed = response.trim();
 		if (trimmed === '') return false;
-		// HTML 错误页通常以 '<' 开头
 		if (trimmed.startsWith('<')) return false;
-		// 既不是 JSON 也不是 JSONP 包装
-		const looksJson = trimmed.startsWith('{') || trimmed.startsWith('[');
-		const looksJsonp = /^\w+\(/.test(trimmed);
-		if (!looksJson && !looksJsonp) return false;
-		return true;
+		return parseJsonp(response) !== response;
```
[the remark this answers is not in the document]
Comment (sansenjian) [reply src/services/apis/y_common.ts]: Fixed in b66f174. Parsed JSON and JSONP objects now pass through the same object validity check, so empty objects trigger the fallback Referer. Added tests for both JSON and JSONP forms.

File: tests/integration/api/api.test.ts [hunk 6]
```diff
@@ -838,6 +838,7 @@ describe('API Integration Tests', () => {
 		});
 
 		test('POST /checkQQLoginQr should return session on success', async () => {
+			const originalUserInfo = global.userInfo;
```
[the remark this answers is not in the document]
Comment (sansenjian) [reply tests/integration/api/api.test.ts]: Updated in b66f174. The test now snapshots the protected userInfo values before login and compares their values afterward, including copied cookie arrays and objects.
